floodlight/filter/filter.py

```diff
@@ -0,0 +1,223 @@
+from typing import Dict
+from floodlight.utils.types import Numeric
+
+import numpy as np
+import scipy.signal
+
+from floodlight import XY
+
+
+def _get_sequences_without_nans(data_column: np.ndarray) -> np.ndarray:
+    """Gets indices of sequences without NaNs from a sequence of data.
+
+    Parameters
+    ----------
+    data_column: np.ndarray
+        One-dimensional array of data with alternating sequences of Numbers and NaNs
+
+    Returns
+    -------
+    sequences: np.ndarray
+        Two-dimensional array with nx2 entrys of n sequences without NaNs and n[0] being
+        the first frame of the sequences and n[1] being the last frame (excluded) of the
+        sequence.
+    """
+
+    # check if even or odd sequences contain NaNs
+    if np.isnan(data_column[0]):
+        first_sequence = 1
+    else:
+        first_sequence = 0
+
+    # indices where nans and numbers are next to each other
+    change_points = np.where(
+        np.diff(np.isnan(data_column), prepend=np.nan, append=np.nan)
+    )[0]
+    # sequences without nans
+    sequences = np.array(
+        [
+            (change_points[i], change_points[i + 1])
+            for i in range(len(change_points) - 1)
+        ]
+    )[first_sequence::2]
```

Comment:
This seems unstable to me, and even if it isn't, it's still quite implicit assuming that change_points returns multiple, alternating sequences -> does that work for empty arrays, array's containing no NaN's or only NaN's?

More explicit would be to explicitly check the initial value of a sequence, and discard those NaN sequences.

Whatever you choose, this portion is probably the most critical one and needs rigorous testing.